### PR TITLE
fix: worker kept open when it shouldn't

### LINKED ISSUE #116: responseObserver.complete() is not called
https://github.com/cloudnc/observable-webworker/blob/f5fe0499ad2ec98dacee2183497fa4e1b9b6bbc1/projects/observable-webworker/src/lib/from-worker.ts#L15

First, thanks for this useful library. When using it for a single web worker, I noticed that my web worker never completes and I had to insert a call to `take(1)` into its pipeline. Looking at your source code, I believe the reason is that `responseObserver.complete()` is never called in the `fromWorker` function - see link.

For the full picture:

Expectation: If I use `fromWorker` and give it an observable which completes after emitting N events as input and use a one-to-one transformation (e.g. the identity function), then the output observable should complete after emitting N events, too.

Actual behavior: The output observable never completes.

Workaround: Insert `take(N)` into the pipeline on the worker side.

### PR DESCRIPTION
**Pull request by maxime1992**
This closes #116

### CODE REVIEW

**Comment by maxime1992** [comment]
I've explained more about the fix here for reference https://github.com/cloudnc/observable-webworker/issues/116#issuecomment-2021396345

File: projects/observable-webworker/src/lib/run-worker.ts [hunk 1]
```diff
@@ -27,8 +27,6 @@ export function getWorkerResult<I, O>(
   const input$ = incomingMessages$.pipe(
     map((e: WorkerMessageNotification<I>): Notification<I> => e.data),
     map((n: Notification<I>) => new Notification(n.kind, n.value, n.error)),
-    // ignore complete, the calling thread will manage termination of the stream
-    filter(n => n.kind !== 'C'),
```
Comment (zakhenry) [inline projects/observable-webworker/src/lib/run-worker.ts:31]: this was clearly intentional, I need to get my head around why.

Comment (maxime1992) [reply projects/observable-webworker/src/lib/run-worker.ts:31]: We've been discussing with @zakhenry privately to understand why the demo worker with the files was working ok.
And Zak found the answer, I'll share here what he found so we remember and anyone interested can understand:

The [issue](https://github.com/cloudnc/observable-webworker/issues/116) reported by @christian-schlichtherle was happening when implementing `DoWork` as shown [here](https://github.com/cloudnc/observable-webworker/issues/116#issuecomment-2020481304).

But there's a split happening here:

https://github.com/cloudnc/observable-webworker/blob/2c9fe413bbba6f245d0399e9cea7eaebfb70a4b9/projects/observable-webworker/src/lib/run-worker.ts#L35-L37

And in the case of the `UnitType` the `materialize` was applied on the `from` :warning:. Therefore, when a `from` was done, it'd emit a complete event, even multiple time if the `concatMap` was kicking in several times. Hence why in this case it was receiving a complete and not in the `DoWork` one.

Still some investigation ongoing though with Zak to dig into the `materialize` atm.

**Comment by christian-schlichtherle** [comment]
Please excuse my insistence, but I think you still need to change the `from-worker.ts` file as shown in my PR. Reason is that the API contract of the interface `Observer` requires a call to either `.error(error)` or `.complete()` to signal the end of the stream. The current code is clearly breaking this contract.

**Comment by zakhenry** [comment]
@christian-schlichtherle it's a little more complicated that that, because with the worker interface we're not dealing with regular observables of values, but with observables of notifications. For this reason we don't expect to call .error or .complete because these signals are present as notifications, and the dematerialize at the end will convert that to the expected observable flow.

I've tried out your fix and unfortunately it breaks the worker pool logic, so we can't accept that PR as-is. We're working on a fix but it might be a few days because this isn't our top priority right now.

**Comment by maxime1992** [comment]
Tested and LGTM

**Pull request merged**